### ISSUE TITLE
Report progressive live streams as dynamic/live to improve UI interactions

### ISSUE THREAD

**Issue by CreeschYT** [google/ExoPlayer#5973]
### Content description
I'm developing an `m3u` player which asks the user to choose an `m3u` file. The `m3u` urls are put in a `ConcatenatingMediaSource` the urls work fine. When the user presses 'Next' the next url should play which works. There are two timers one on the left and on the right. The one on the left counts. The one on the right stays at zero (because of `m3u livestreaming`). The problem is that when I press 'Previous' the counter on the left goes to 0:00 and then back to the time before the Previous button was pressed. The url being played has not changed. So I have to click to Previous button twice and fast to go back instead of pressing once.  The Code:

```
try {
                BufferedReader br = new BufferedReader(new FileReader(filePath));
                String line;

                while ((line = br.readLine()) != null) {
                   if (line.startsWith("h")){
                       lineStore.add(line);

                   }

                }
                br.close();
            } catch (Exception e) {
                Log.e("MainAcvtivity"," exoplayer error "+ e.toString());
            }

            for (String url : lineStore) {
                mediaSources.add(new ExtractorMediaSource.Factory(dataSourceFactory).createMediaSource(Uri.parse(url)));

            }

            concatenatingMediaSource = new ConcatenatingMediaSource(mediaSources.toArray(new MediaSource[mediaSources.size()]));
            exoPlayer.prepare(concatenatingMediaSource);
            exoPlayer.setPlayWhenReady(true);

```

### Link to test content
The m3u file I used:
[Test.zip](https://github.com/google/ExoPlayer/files/3237525/Test.zip)

### Version of ExoPlayer being used
I'm currently using exoplayer 2.9.1

### Device(s) and version(s) of Android being used
I expected my app to run on a minimum of Android version 5.0. I also have tested my App on my phone (Samsung S6, Android 7.0) and on a (Samsung s4 mini, Android 6.0)

**Comment by ojw28** [comment]
This is working as intended for on-demand streams and for live streams that are seekable. In these cases pressing back will seek back to the start of the current stream, unless playback is already near the start.

You're right that this doesn't make sense for non-seekable live streams, which is what you're dealing with here. We should adjust the logic for that case.

**Comment by CreeschYT** [comment]
umm. Excuse me, but why did you add the label low priority ?

**Comment by CreeschYT** [comment]
Can you atleast give me a reason that this is low priority ?

**Comment by tonihei** [comment]
Actually, the logic mentioned above is already implemented. The underlying problem here is that progressive live streams are not marked as "dynamic". Marking as an enhancement instead. 

Note that this is also related to #2668 where we try to improve the way to denote live streams in general.

The "low priority" label is used to mark issues which we not plan to address immediately. These are often things which are too specialized or require major changes. Removing the label for this case.

**Comment by CreeschYT** [comment]
Ok, thank you

**Comment by ojw28** [comment]
@tonihei - Can this be closed as fixed?

**Comment by tonihei** [comment]
No, the second part about improving UI interactions isn't solved yet.
